src/mono/wasm/debugger/BrowserDebugProxy/EvaluateExpression.cs

```diff
@@ -185,96 +190,73 @@ public SyntaxTree ReplaceVars(SyntaxTree syntaxTree, IEnumerable<JObject> ma_val
                 {
                     case "string":
-                        return value?.Value<string>();
+                    {
+                        var str = value?.Value<string>();
+                        str = str.Replace("\"", "\\\"");
+                        valueRet = $"\"{str}\"";
+                        typeRet = valueRet.GetType().FullName;
+                        break;
+                    }
                     case "number":
-                        return value?.Value<double>();
+                        valueRet = value?.Value<double>();
+                        typeRet = valueRet.GetType().FullName;
+                        break;
                     case "boolean":
-                        return value?.Value<bool>();
+                        valueRet = value?.Value<string>().ToLower();
+                        typeRet = "System.Boolean";
+                        break;
                     case "object":
-                        return variable;
+                        valueRet = "Newtonsoft.Json.Linq.JObject.FromObject(new {"
+                            + $"type = \"{type}\""
+                            + $", description = \"{variable["description"].Value<string>()}\""
+                            + $", className = \"{variable["className"].Value<string>()}\""
+                            + (subType != null ? $", subtype = \"{subType}\"" : "")
+                            + (objectId != null ? $", objectId = \"{objectId}\"" : "")
+                            + "})";
+                        typeRet = "object";
+                        break;
                     case "void":
-                        return null;
-                }
-                throw new Exception($"Evaluate of this datatype {type} not implemented yet");//, "Unsupported");
-            }
-
-            private string GetTypeFullName(JToken variable)
-            {
-                string type = variable["type"].ToString();
-                string subType = variable["subtype"]?.Value<string>();
-                object value = ConvertJSToCSharpType(variable);
-
-                switch (type)
-                {
-                    case "object":
-                        {
-                            if (subType == "null")
-                                return variable["className"].Value<string>();
-                            else
-                                return "object";
-                        }
-                    case "void":
-                        return "object";
+                        valueRet = "Newtonsoft.Json.Linq.JObject.FromObject(new {"
+                                + $"type = \"object\","
+                                + $"description = \"object\","
+                                + $"className = \"object\","
+                                + $"subtype = \"null\""
+                                + "})";
+                        typeRet = "object";
+                        break;
```

Comment:
I'm not sure I understand these two.
1. why do we need to use JObjects in the script?
2. `case "object":` - would it be better to use `Newtonsoft.Json.Linq.JObject.Parse(..)` and pass the escaped string for `variable`?
3. `case "void":` - why is this needed? And what does the `variable` look like here? And do we have a test for this?

Reply:
1) Because that is how we return simple variable values, so when it's something that we really need to execute we also keep the same way of returning the value.
2) I think it's faster to not parse the string? once this is already parsed we don't need to parse it again?
3) Yes, the tests evaluating method that return void.